### DISCUSSION TITLE
Feature Request: Input Volume Control

### DISCUSSION THREAD

**Discussion by domsch1988** [File-New-Project/EarTrumpet#691] [Ideas]
I think this has been touched on in other Issues, but i wanted to bring this up again.
At my PC i have 4 different Input devices (Audio Interface, USB Mic, Headset). Certain Programs (Especially games) tend to set the Input levels for several audio devices to a really low value and don't return it to 100% after that.

Having a quick way to get the exact UI we now have for outputs, but for inputs would be epic. Maybe a Ctrl+Click or Shift+Click on the tray icon to get the Input Mixer. If it did include the Peak Meter that's be extra awesome. But having a quick way to set my mic back to 100% Level would already save me from digging through the settings app multiple times a day.

Not sure if wanted/needed, but i'd be willing to pay/donate for such a feature.

**Comment by domsch1988** [comment]
Ok, One and a half years later i'm coming back to this. Microsoft is burrying the Setting for Input Volumes deeper and deeper and i have not found any other tool that would let me change the Microphones Volume from the tray.

So instead of sitting on my lazy butt asking for a feature again, i decided to take a look at the source instead and see what i can do. I have close ot zero Experience in C# but better than nothing.
To my surprise, the Device kind "Recording" already exists and is close to functional. I was able to build a version that works exclusively for input devices. While that might not be what most people want, it's good enough for me.

It has some issues though. One can't switch input devices that way, There is no seperate icon present and some other minor things. Granted, i didn't dive too deep and my understanding of the code is still exceptionally shallow. All i basically did was tell Eartrumpet to initiallize Input devices, rather than Playback devices.
I still have no clue how to get both working, as this seems to be an either/or condition. For my specific usecase, this works 100% better than nothing. I rarely need Playback Volume Controll, so having i seperate EarTrumpet build just for Recording devices works fine.

I'll probably use this opportunity to dive deeper into C# and try to understand what's going on here to maybe get this to a more functional Version. I have no idea if this feature is in consideration, but from the code it looks like it is. I have a workaround know and will be looking closely at how this keeps developing.

**Reply by davuxcom** [reply]
I'm always impressed by people who will actually look at the code. Thanks for diving in. We started eyeing an addon called "MouthTrumpet" which is why everything is so nicely set up to take the same audio manager object and pass it down. :)

It doesn't really work that well since microphones don't have apps, and some other problems I don't recall at the moment.

Glad you got it working though.

**Reply by argumemnon** [reply]
Hey @domsch1988 - any chance you can publish a fork to your own repo?
